### DISCUSSION TITLE
Transpose chord to another

### DISCUSSION THREAD

**Discussion by zipzapduo** [melanchall/drywetmidi#138] [Q&A]
my main goal here is to transpose all chords from the collection to a particular one with scale correction. Is it possible?

**Comment by melanchall** [comment]
Can you please provide an example? Do you mean transpose a chord X by N steps within scale S? So we need two parameters: the number of scale steps and scale?

**Comment by zipzapduo** [comment]
So let's say we have a midi part in dmin with chords Amin, G, Dmin, C. I want to transpose all chords different from Dmin. In this case Amin, G and C. The final result is a pattern in Dmin. Also the transpose has to be iwith scale correction . In this case Dmin

**Reply by zipzapduo** [reply]
Basically what I want is to transpose tones or single notes. Examples:

A chord of Amin with notes C,E,A will transform to F,A,D in Dmin 

A single note G in Amin it will be C in Dmin .

Does it make sense ?

I think that what is missing to me is a way of calculating the interval between one tone to another

**Reply by melanchall** [reply]
Well, unfortunately there is no dedicated API in DryWetMIDI to handle your task at now. You can do this manually implementing music theory algorithms in your program. This is an interesting feature though so I'll think about how to implement it in the library. But I can't say any date when it will be availabale. I suppose in the next year.

**Comment by zipzapduo** [comment]
Ok. I will try to implement with what the API does but I think it is too much difficult.I will dedicate some attention on future releases
